snippets/country-localization.liquid

```diff
@@ -5,41 +5,143 @@
     - localPosition: pass in the position in which the form is coming up to create specific IDs
 {%- endcomment -%}
 
+{%- liquid
+  assign currencies = localization.available_countries | map: 'currency' | map: 'iso_code' | uniq
+  assign popular_countries = localization.available_countries | where: 'popular?' | sort: 'name'
+
+  assign show_country_filter = false
+  if localization.available_countries.size > 9
+    assign show_country_filter = true
+  endif
+
+  assign show_popular_countries = false
+  if localization.available_countries.size > 9 and popular_countries.size > 1
+    assign show_popular_countries = true
+  endif
+%}
+
 <div class="disclosure">
   <button
     type="button"
     class="disclosure__button localization-form__select localization-selector link link--text caption-large"
     aria-expanded="false"
-    aria-controls="{{ localPosition }}List"
+    aria-controls="{{ localPosition }}-country-results"
     aria-describedby="{{ localPosition }}Label"
   >
     <span>
-      {{- localization.country.currency.iso_code }}
-      {{ localization.country.currency.symbol }} | {{ localization.country.name -}}
+      {{- localization.country.name }} |
+      {{ localization.country.currency.iso_code }}
+      {{ localization.country.currency.symbol -}}
     </span>
     {% render 'icon-caret' %}
   </button>
-  <div class="disclosure__list-wrapper" hidden>
-    <ul id="{{ localPosition }}List" role="list" class="disclosure__list list-unstyled">
-      {%- for country in localization.available_countries -%}
-        <li class="disclosure__item" tabindex="-1">
-          <a
-            class="link link--text disclosure__link caption-large focus-inset{% if country.iso_code == localization.country.iso_code %} disclosure__link--active{% endif %}"
-            href="#"
-            {% if country.iso_code == localization.country.iso_code %}
-              aria-current="true"
-            {% endif %}
-            data-value="{{ country.iso_code }}"
+  <div class="disclosure__list-wrapper country-selector" hidden>
+    <div class="country-selector__actions {% unless show_country_filter %}no-padding{% endunless %}">
```

Comment:
I think a class name that would be a "modifier" following BEM convention would be better and less generic. 

```suggestion
    <div class="country-selector__actions {% unless show_country_filter %}country-selector__actions--no-padding{% endunless %}">
```

It's definitely longer but at least not as generic. If it was a class we use in different spots for the same outcome it would make sense but for now I think it's only used here.